targets/TARGET_ARM_SSG/TARGET_CM3DS_MPS2/serial_api.c

```diff
@@ -0,0 +1,378 @@
+        case UART_3: CMSDK_UART2->BAUDDIV = baudrate_div; break;
+        default: error("serial_baud"); break;
+    }
+    } else {
+        error("serial_baud");
+    }
+
+}
+
+void serial_format(serial_t *obj, int data_bits, SerialParity parity, int stop_bits) {
+}
+
+/******************************************************************************
+ * INTERRUPTS HANDLING
+ ******************************************************************************/
+static inline void uart_irq(uint32_t intstatus, uint32_t index, CMSDK_UART_TypeDef *puart) {
+    SerialIrq irq_type;
+    switch (intstatus)
+    {
+        case 1:
+        {
+            irq_type = TxIrq;
+        }
+        break;
+
+        case 2:
+        {
+            irq_type = RxIrq;
+        }
+        break;
+
+        default: return;
+    }   /* End of Switch */
+
+    if ((RxIrq == irq_type) && (NC != uart_data[index].sw_rts.pin))
+    {
+        gpio_write(&uart_data[index].sw_rts, 1);
+        // Disable interrupt if it wasn't enabled by other part of the application
+        if (!uart_data[index].rx_irq_set_api)
+        {
+            /* Disable Rx interrupt */
+            puart->CTRL &= ~(CMSDK_UART_CTRL_RXIRQEN_Msk);
+        }
+    }
+
+    if (uart_data[index].serial_irq_id != 0)
+    {
+        if ((irq_type != RxIrq) || (uart_data[index].rx_irq_set_api))
+        {
+            irq_handler(uart_data[index].serial_irq_id, irq_type);
+        }
+    }
+
+    if( irq_type == TxIrq )
+    {
+        /* Clear the TX interrupt Flag */
+        puart->INTCLEAR |= 0x01;
+    }
+    else
+    {
+        /* Clear the Rx interupt Flag */
+        puart->INTCLEAR |= 0x02;
+    }
+}
+
+void uart0_irq() {uart_irq(CMSDK_UART1->INTSTATUS & 0x3, 0, (CMSDK_UART_TypeDef*)CMSDK_UART1);}
+void uart1_irq() {uart_irq(CMSDK_UART3->INTSTATUS & 0x3, 1, (CMSDK_UART_TypeDef*)CMSDK_UART3);}
+void uart2_irq() {uart_irq(CMSDK_UART0->INTSTATUS & 0x3, 2, (CMSDK_UART_TypeDef*)CMSDK_UART0);}
+void uart3_irq() {uart_irq(CMSDK_UART2->INTSTATUS & 0x3, 3, (CMSDK_UART_TypeDef*)CMSDK_UART2);}
+
+void serial_irq_handler(serial_t *obj, uart_irq_handler handler, uint32_t id) {
+    irq_handler = handler;
+    uart_data[obj->index].serial_irq_id = id;
+}
+
+static void serial_irq_set_internal(serial_t *obj, SerialIrq irq, uint32_t enable) {
+    IRQn_Type irq_n = (IRQn_Type)0;
+    uint32_t vector = 0;
+    switch ((int)obj->uart)
+    {
+        case UART_0:
+            irq_n = UART0_IRQn;
+            vector = (uint32_t)&uart0_irq;
+        break;
+        case UART_1:
+            irq_n = UART1_IRQn;
+            vector = (uint32_t)&uart1_irq;
+        break;
+        case UART_2:
+        {
+            irq_n = UART2_IRQn;
+            vector  = (uint32_t)&uart2_irq;
+        }
+        break;
+        case UART_3:
+        {
+            irq_n = UART3_IRQn;
+            vector  = (uint32_t)&uart3_irq;
+        }
+        break;
+    }
+
+    if (enable)
```

Comment:
https://docs.mbed.com/docs/mbed-os-handbook/en/5.1/cont/code_style/ - should be ``if () {``